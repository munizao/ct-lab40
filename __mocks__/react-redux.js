jest.mock('react-redux', () => {
  return {
    __esModule: true,
    useDispatch: jest.fn(() => () => {}),
    useSelector: jest.fn((selector) => selector({ user: {}, repos: {} })),
  };
});
