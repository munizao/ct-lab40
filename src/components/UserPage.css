html, body {
  height: 100vh;
  margin: 0px;
  font-family: Arial, Helvetica, sans-serif;
  color: white;
  background-color: #613d07;
}

.UserPage {
  header {
    padding: 1rem;
    h1 {
      margin: 0px;
    }
  }
  main {
    padding: 1rem;
    .Controls {
      input {
        margin: 0px .7rem .7rem 0px;
      }
    }
    .Results {
      display: flex;
      flex-direction: row;
    }
  }
}
