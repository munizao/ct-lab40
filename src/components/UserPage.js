import React, { useState } from 'react';
import { fetchUser } from '../redux/actions/actions';
import { useDispatch } from 'react-redux';
import UserDisplay from './UserDisplay/UserDisplay';
import ReposDisplay from './ReposDisplay/ReposDisplay';
import styles from './UserPage.css';


const UserPage = () => {
  const [userName, setUserName] = useState('');
  const dispatch = useDispatch();

  return (
    <>
      <header>
        <h1>GitHub User Search</h1>
      </header>
      <main className={styles.Main} >
        <section className={styles.Controls}>
          <input type="text" value={userName}  onChange={({ target }) => setUserName(target.value)} />
          <button onClick={() => dispatch(fetchUser(userName))}>Submit</button>
        </section>
        <section className={styles.Results}>
          <UserDisplay />
          <ReposDisplay />
        </section>
      </main>
    </>
  );
};

export default UserPage;
