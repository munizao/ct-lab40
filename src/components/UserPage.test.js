import React from 'react';
import 'react-redux';
import { shallow } from 'enzyme';
import UserPage from './UserPage';

describe('UserPage component', () => {
  it('renders UserPage', () => {
    const wrapper = shallow(<UserPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
