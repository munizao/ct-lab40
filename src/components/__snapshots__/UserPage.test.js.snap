// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage component renders UserPage 1`] = `
<div
  className="UserPage"
>
  <header>
    <h1>
      GitHub User Search
    </h1>
  </header>
  <main
    className="Main"
  >
    <section
      className="Controls"
    >
      <input
        onChange={[Function]}
        placeholder="GitHub User"
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </section>
    <section
      className="Results"
    >
      <UserDisplay />
      <ReposDisplay />
    </section>
  </main>
</div>
`;
